components/script/dom/permissions.rs

```diff
@@ -0,0 +1,243 @@
+            Ok(ConversionResult::Success(descriptor)) => Ok(descriptor),
+            Ok(ConversionResult::Failure(message)) => Err(String::from(message)),
+            Err(_) => Err(String::from(BT_DESC_CONVERSION_ERROR)),
+        }
+    }
+}
+
+// https://w3c.github.io/permissions/#dom-permissions-query
+fn sync_default_permission_query_call(global: &GlobalScope,
+                                      descriptor: PermissionDescriptor,
+                                      promise: &Rc<Promise>,
+                                      cx: *mut JSContext) {
+    // Step 5.
+    let status = PermissionStatus::create_from_descriptor(global, PermissionDescriptorType::Default(descriptor));
+
+    // Step 6.
+    // NOTE: `status.query` is the same as descriptor
+    status.permission_query(&*status.get_query().borrow());
+
+    // Step 7.
+    promise.resolve_native(cx, &status);
+}
+
+// https://w3c.github.io/permissions/#dom-permissions-request
+fn sync_default_permission_request_call(global: &GlobalScope,
+                                        descriptor: PermissionDescriptor,
+                                        promise: &Rc<Promise>,
+                                        cx: *mut JSContext) {
+    // Step 5.
+    let status = PermissionStatus::create_from_descriptor(global, PermissionDescriptorType::Default(descriptor));
+
+    // Step 6.
+    // NOTE: `status.query` is the same as descriptor
+    if let Err(err) = status.permission_request(&*status.get_query().borrow()) {
+        // Step 7.
+        return promise.reject_error(cx, err);
+    }
+
+    // Step 8.
+    promise.resolve_native(cx, &status);
+}
+
+// https://w3c.github.io/permissions/#dom-permissions-query
+fn bluetooth_permission_query_call(global: &GlobalScope,
+                                   descriptor: BluetoothPermissionDescriptor,
+                                   promise: &Rc<Promise>,
+                                   cx: *mut JSContext) {
+    // Step 5.
+    let result = BluetoothPermissionResult::create_from_descriptor(
+                    global, PermissionDescriptorType::Bluetooth(descriptor));
+
+    // Step 6.
+    result.permission_query(&promise, cx);
+}
+
+// https://w3c.github.io/permissions/#dom-permissions-request
+fn bluetooth_permission_request_call(global: &GlobalScope,
+                                     descriptor: BluetoothPermissionDescriptor,
+                                     promise: &Rc<Promise>,
+                                     cx: *mut JSContext) {
+    // Step 5.
+    let result =
+        BluetoothPermissionResult::create_from_descriptor(global, PermissionDescriptorType::Bluetooth(descriptor));
+    // Step 6.
+    if let Err(err) = result.permission_request(&promise) {
+        // Step 7.
+        return promise.reject_error(cx, err);
+    }
+    // NOTE: Step 8. is in `bluetoothpermissionresult.rs` in the `handle_response` function.
```

Comment:
I don't like these "find the rest in file X at Y" comments; things tend to get renamed quite often.